projects/scannet_offline_eval/build_sparse_voxel_map.py

```diff
@@ -0,0 +1,475 @@
+    # 2D map projections for planning
+    ##############################################
+    def get_2d_map(self):
+        """Get 2d obstacle map for low level motion planning and frontier-based exploration"""
+        return self.voxel_map.get_2d_map()
+
+    def show(self) -> Tuple[np.ndarray, np.ndarray]:
+        """Display the aggregated point cloud."""
+        return self.voxel_map.show(
+            instances=True,
+            height=1000,
+            boxes_plot_together=False,
+            boxes_name_int_to_display_name_dict=dict(
+                enumerate(self.metadata.thing_classes)
+            ),
+            backend="pytorch3d",
+        )
+
+
+def compress_instance_bounds_(
+    instances: Sequence[Instance],
+    drop_prop: float,
+    voxel_size: float,
+    min_voxels_to_compress: int = 3,
+    min_vol: float = 1e-6,
+):
+    """Trailing _ in torch indicate in-place"""
+    for instance in instances:
+        reduced_points = drop_smallest_weight_points(
+            instance.point_cloud,
+            drop_prop=drop_prop,
+            voxel_size=voxel_size,
+            min_points_after_drop=min_voxels_to_compress,
+        )
+        new_bounds = get_bounds(reduced_points)
+        if box3d_volume_from_bounds(new_bounds) >= min_vol:
+            instance.bounds = new_bounds
+    return instances
+
+
+def drop_smallest_weight_points(
+    points: Tensor,
+    voxel_size: float = 0.01,
+    drop_prop: float = 0.1,
+    min_points_after_drop: int = 3,
+):
+    voxel_pcd = VoxelizedPointcloud(
+        voxel_size=voxel_size,
+        dim_mins=None,
+        dim_maxs=None,
+        feature_pool_method="mean",
+    )
+    voxel_pcd.add(
+        points=points,
+        features=None,  # instance.point_cloud_features,
+        rgb=None,  # instance.point_cloud_rgb,
+    )
+    orig_points = points
+    points = voxel_pcd._points
+    weights = voxel_pcd._weights
+    assert len(points) > 0, points.shape
+    weights_sorted, sort_idxs = torch.sort(weights, dim=0)
+    points_sorted = points[sort_idxs]
+    weights_cumsum = torch.cumsum(weights_sorted, dim=0)
+    above_cutoff = weights_cumsum >= (drop_prop * weights_cumsum[-1])
+    cutoff_idx = int(above_cutoff.max(dim=0).indices)
+    if len(points_sorted[cutoff_idx:]) < min_points_after_drop:
+        return orig_points
+    # print(f"Reduced {len(orig_points)} -> {len(points)} -> {above_cutoff.sum()}")
+    return points_sorted[cutoff_idx:]
+
+
+if __name__ == "__main__":
+    from evaluation.obj_det import eval_bboxes_and_print
+
+    from home_robot.perception.detection.detic.detic_perception import DeticPerception
+
+    parser = argparse.ArgumentParser(
+        prog="build_3d_map",
+        description="Builds 3D map of a scannet scene from a RGBD trajectory and evaluates object detection",
+    )
+
+    parser.add_argument(
+        "--scannet-dir",
+        default="/private/home/ssax/home-robot/src/home_robot/home_robot/datasets/scannet/data",
```

Comment:
hard coded path? can we do a relative path instead? usually i have these coded to assume youre running from $HOME_ROBOT_ROOT